client/src/components/pages/personalProfile/index.js

```diff
@@ -0,0 +1,187 @@
+import React from 'react'
+import { Form, Input } from 'antd'
+import Button from '../../utils/Button'
+
+import 'antd/dist/antd.css'
+import './style.css'
+
+class PersonalForm extends React.Component {
+
+  state = {
+    confirmDirty: false,
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.form.validateFieldsAndScroll((err, values) => {
+      if (!err) {
+        console.log('Received values of form: ', values);
+      }
+    });
+  };
+
+  handleConfirmBlur = e => {
+    const value = e.target.value;
+    this.setState({ confirmDirty: this.state.confirmDirty || !!value });
+  };
+
+  compareToFirstPassword = (rule, value, callback) => {
+    const form = this.props.form;
+    if (value && value !== form.getFieldValue('password')) {
+      callback('Password must match!');
+    } else {
+      callback();
+    }
+  };
+
+  validateToNextPassword = (rule, value, callback) => {
+    const form = this.props.form;
+    if (value && this.state.confirmDirty) {
+      form.validateFields(['confirm'], { force: true });
+    }
+    callback();
+  };
+
+  render() {
+    const { getFieldDecorator } = this.props.form;
+
+    const formItemLayout = {
+      labelCol: {
+        xs: { span: 0 },
+        sm: { span: 0 },
+      },
+      wrapperCol: {
+        xs: { span: 0 },
+        sm: { span: 24 },
+      },
+    };
+    const tailFormItemLayout = {
+      wrapperCol: {
+        xs: {
+          span: 24,
+          offset: 0,
+        },
+        sm: {
+          span: 16,
+          offset: 8,
+        },
+      },
+    };
+
+    return (
+      <Form {...formItemLayout} onSubmit={this.handleSubmit} className='form'>
+
+        <Form.Item label="First Name" className='form--item'>
+          {getFieldDecorator('first', {
+            rules: [
+              {
+                type: 'string',
+                message: 'The First Name should be string!',
+              },
+              {
+                whitespace: true,
+                message: 'Delete the spaces!'
```

Comment:
@AbdallahAmmar96 I know someone that has a space in their first name 😂 so maybe this rule is too strict